rfcs/0146-meta-categories.md

```diff
@@ -0,0 +1,354 @@
+   be precisely the opposite: Nixpkgs being _the_ source of structured metainfo
+   for other software collections.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Do nothing
+
+   This will exacerbate the problems already listed.
+
+2. Ignore/nuke the categorization completely
+
+   This is an alternative worthy of some consideration. After all,
+   categorization is not without its problems, as shown above. Removing or
+   ignoring classification removes all problems.
+
+   However, there are good reasons to keep the categorization:
+
+   - The complete removal of categorization is too harsh. A solution that keeps
+     and enhances the categorization is way more preferrable than one that nukes
+     it completely.
+
+   - As said before, the categorization is already present; this RFC proposes to
+     expose it to a higher level, in a structured, more discoverable format.
+
+   - Categorization is very traditional among software collections. Many of them
+     are doing this just fine for years on end, and Nixpkgs can imitate them
+     easily - and even surpass them, given the benefits of Nix language
+     machinery.
+
+   - Categorization is useful in many scenarios and use cases - indeed they
+     are ubiquitous in software world:
+     - specialized search engines (from Repology to MELPA)
+     - code forges, from Sourceforge to Gitlab
+     - as said above, software collections from pkgsrc to slackbuilds
+     - organization and preservation (as Software Heritage)
+
+3. Debtags/Appstream hybrid approach
+
+A hybrid approach for code implementation would be implement two meta
+attributes, namely
+
+- `meta.categories` for Appstream-based categories
+  -  the corresponding `lib.categories` should follow Appstream closely, with
+     few room to custom/extra categories
+- `meta.tags` for Debtags-like tags
+  - while being inspired from the venerable Debtags work, the corresponding
+    `lib.tags` is completely free to modify and even divert from Debtags,
+    following its own way
+- generally speaking, `lib.tags` should be less bureaucratic than
+  `lib.categories`
+
+However, this approach arguably elevates the complexity of the whole work, and
+adds too much redundancy.
+
+# Prior art
+[prior-art]: #prior-art
+
+As said above, categorization is very traditional among software collections. It
+is not hard to cite examples in this arena; the most interesting ones I have
+found are listed below (linked at [references section](#references)):
+
+- FreeBSD Ports;
+- Debtags;
+- Appstream Project;
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+There are some still unsolved issues:
+
+- What data structure is suitable to represent a category?
+
+  - For now we stick to the most natural: a set `{ name, description }`.
+
+- Should we have a set of primary, "most important" categories with mandatory
+  status, in the sense each package should set at least one of them?
+  - The answer is most certainly positive.
```

Comment:
I'd appreciate it if the RFC made it clear if it expects these questions to be answered by the categorization team or if it reserves them for future RFCs.

My personal take is the first question should be left to the team and the second should be answered by this RFC.

Reply:
Arguably, the current text of the entire RFC gives the team authority to resolve both as possibilities of implementation, but puts a strong preference on the choice in the second one.

What exactly does it mean «each package» for primary categories in some corner cases will probably have to be refined iteratively, though, so the team gets some freedom on refining the exact interpretation without going against the letter of the RFC.

I agree that it is better to spell out that the issues are for the team.

Reply:
Why not both?
The categorization team decides whether it can be solved right away or if it deserves another RFC.